api/healthcheck/healthcheck_controller.go

```diff
@@ -17,32 +17,38 @@
 package healthcheck
 
 import (
+	h "github.com/InVisionApp/go-health"
+	"github.com/Peripli/service-manager/pkg/health"
 	"github.com/Peripli/service-manager/pkg/util"
 	"net/http"
 
-	"github.com/Peripli/service-manager/pkg/health"
 	"github.com/Peripli/service-manager/pkg/log"
 	"github.com/Peripli/service-manager/pkg/web"
 )
 
 // controller platform controller
 type controller struct {
-	indicator health.Indicator
+	health           h.IHealth
+	aggPolicy        health.AggregationPolicy
+	failuresTreshold int64
 }
 
 // NewController returns a new healthcheck controller with the given indicators and aggregation policy
-func NewController(indicators []health.Indicator, aggregator health.AggregationPolicy) web.Controller {
+func NewController(health h.IHealth, aggPolicy health.AggregationPolicy, failuresTreshold int64) web.Controller {
 	return &controller{
-		indicator: newCompositeIndicator(indicators, aggregator),
+		health:           health,
+		aggPolicy:        aggPolicy,
+		failuresTreshold: failuresTreshold,
 	}
 }
 
 // healthCheck handler for GET /v1/monitor/health
 func (c *controller) healthCheck(r *web.Request) (*web.Response, error) {
 	ctx := r.Context()
 	logger := log.C(ctx)
-	logger.Debugf("Performing health check with %s...", c.indicator.Name())
-	healthResult := c.indicator.Health()
+	logger.Debugf("Performing health check...")
+	healthState, _, _ := c.health.State()
```

Comment:
Here second argument is overall status of health, which is not needed since we have custom logic for aggregation, and the error is the potential error for overall health failure (error in some of the indicators). This error is not literally ignored because it is presented in the state of the concrete indicator which failed and is processed in aggregation (included into response).